pkg/trait/prometheus.go

```diff
@@ -116,6 +82,67 @@ func (t *prometheusTrait) Apply(e *Environment) (err error) {
 	return nil
 }
 
+func (t *prometheusTrait) configurePorts(e *Environment) bool {
+	service, servicePort := t.configureServicePort(e)
+	container, containerPort := t.configureContainerPort(e)
+
+	condition := v1alpha1.IntegrationCondition{
+		Type: v1alpha1.IntegrationConditionPrometheusAvailable,
+	}
+
+	if containerPort == nil {
+		condition.Status = corev1.ConditionFalse
+		condition.Reason = v1alpha1.IntegrationConditionContainerNotAvailableReason
+	} else {
+		container.Ports = append(container.Ports, *containerPort)
+
+		condition.Status = corev1.ConditionTrue
+		condition.Reason = v1alpha1.IntegrationConditionPrometheusAvailableReason
+		condition.Message = fmt.Sprintf("%s(%s/%d)", container.Name, containerPort.Name, containerPort.ContainerPort)
+	}
+
+	if servicePort != nil {
```

Comment:
Shouldn't the condition be updated with the reason the service is not available when `service` is `nil`?